### ISSUE TITLE
Cross-node routes are lost after all RR nodes are powered off

### ISSUE THREAD

**Issue by gongzixiangyuan** [projectcalico/calico#9259]
1. I have a Kubernetes cluster with six nodes, three control plane nodes, and three worker nodes.
2. I select master1 and master2 on the control plane as route reflectors.
3. Power off master 1 and master 2.
4. Cross-node routes on all nodes that are not powered off are lost.

## Expected Behavior

Remaining nodes can run inertly, and cross-node communication between existing pods is not affected.

## Current Behavior

Cross-node routes on all nodes that are not powered off are lost, and cross-node communication between existing pods fails.

## Possible Solution

When the calico-node pod is deleted, the bird receives the SIGTERM signal and performs graceful shutdown. As a result, the bird routes on the remaining nodes are lost. 
- bird log: **Received: Administrative shutdown**
Ignoring the SIGTERM signal is fine.

## Steps to Reproduce (for bugs)

1. Power off master 1 and master 
2. Cross-node routes on all nodes that are not powered off are lost.

## Your Environment

Calico version 3.27.3
Orchestrator version (e.g. kubernetes, mesos, rkt): 1.28.1

**Comment by gongzixiangyuan** [comment]
It seems that maxRestartTime expires after more than 60 minutes. The long lived stale time of bird is affected. The default value is 3600s.

Can I add another configuration item to bird.cfg.template? For example:
`{{- if ne $data.restart_time ""}}
graceful restart time {{$data.restart_time}};
long lived stale time {{$data.restart_time}};
{{- end}}
`

**Comment by gongzixiangyuan** [comment]
```shell
{{- if ne $data.restart_time ""}}
graceful restart time {{$data.restart_time}};  # Add this line
long lived stale time {{$data.restart_time}};  # Add this line
{{- end}}
```

Adding the above two lines seems to break the 3600s limit.

**Comment by mazdakn** [comment]
@gongzixiangyuan IIUC, in your cluster:
- 2 master nodes act as RR.
- All nodes are peered with RRs.
- Now you want to power off the master/RR nodes for a long period (more than 1 hour)
- And see cross node traffic continue to work?

Can you explain what's the use case and what are you trying to achieve?

**Comment by mazdakn** [comment]
Also why do you need RR for such a small cluster?
1 - A BGP mesh would be easier and shutting down masters node won't affect cross cluster traffic.
2 - Using an overlay like VXLAN, you won't need BGP in first place.

**Comment by gongzixiangyuan** [comment]
@mazdakn  

Thank you for your reply! As you understand it.
But, in my real-life scenario, it's going to be over 100 nodes. What I want to verify is that the control plane failure does not affect the inertial operation of the data plane.

**Comment by coutinhop** [comment]
@gongzixiangyuan I think your issue is inherently caused by the way BGP and RRs work, in that the RR(s) is(are) point(s) of failure. If all of your RRs are down, then your routes will be lost eventually.

What exactly are you trying to achieve? Resilience while a RR restarts? 3600s seems like enough time for that, you could have other problems allowing stale routes for much longer than that...

But with the understanding that we intrinsically need at least one working RR for our topology to be healthy, I'd advice to add more RRs for redundancy, and to be aware that they can't all be turned off at once.

**Comment by gongzixiangyuan** [comment]
@coutinhop  I hope that the container network on other nodes will not be affected even if all RR nodes are powered down, which seems to be contrary to the original design intention of BGP

**Comment by caseydavenport** [comment]
I don't think we can expect routes to remain intact if all RR nodes are powered down indefinitely. I think the way this problem is intended to be solved is with redundancy in route reflector topology.

We could arbitrarily increase the timer, as you mentioned, but that has the side effect of leaving potentially bad routing information in place, so I would not recommend solving this by just increasing the hold timers to a large number.

For anyone looking at this, I would double check that you actually need BGP. Calico's VXLAN does the job and doens't have these topological complexities. The reason for enabling Calico BGP should largely be integration with the rest of your network.

If you do need BGP, then configuring redundancy in your topology is probably the way to handle transient failures / node reboots.

**Comment by gongzixiangyuan** [comment]
> I don't think we can expect routes to remain intact if all RR nodes are powered down indefinitely. I think the way this problem is intended to be solved is with redundancy in route reflector topology.
> 
> We could arbitrarily increase the timer, as you mentioned, but that has the side effect of leaving potentially bad routing information in place, so I would not recommend solving this by just increasing the hold timers to a large number.
> 
> For anyone looking at this, I would double check that you actually need BGP. Calico's VXLAN does the job and doens't have these topological complexities. The reason for enabling Calico BGP should largely be integration with the rest of your network.
> 
> If you do need BGP, then configuring redundancy in your topology is probably the way to handle transient failures / node reboots.

Got it, thank you!
